composeApp/src/main/kotlin/io/github/couchtracker/utils/Loadable.kt

```diff
@@ -1,39 +1,71 @@
 package io.github.couchtracker.utils
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 /**
  * Extension of [Result] that allows the value to be [Loading].
  */
-sealed interface Loadable<out T, out E> : Actionable<T, E> {
+sealed interface Loadable<out T> : Actionable<T, Nothing> {
```

Comment:
If we are uncoupling `Loadable` from `Result`, can we also do the same for `Actionable`. i.e. `Actionable` can be of a generic `T` and we can have functions to deal with an `Actionable<Loadable>`